### PR TITLE
add tipProps on tooltipslider

### LINKED ISSUE #276: rc-tooltip in a react bootstrap Modal
Hi, 
I got a issue when using the rc-tooltip in a react bootstrap Modal whereby the tooltip is behind the modal and therefore not visible.
Any idea on how to fix that?

### PR DESCRIPTION
**Pull request by zeakd**
After createSliderWIthTooltip, slider can get tipProps.  close #276

### CODE REVIEW

File: src/createSliderWithTooltip.jsx [hunk 1]
```diff
@@ -26,14 +28,26 @@ export default function createSliderWithTooltip(Component) {
...
         <Tooltip
-          prefixCls="rc-slider-tooltip"
-          overlay={tipFormatter(value)}
           visible={!disabled && (this.state.visibles[index] || dragging)}
-          placement="top"
           key={index}
+          prefixCls={prefixCls}
+          overlay={overlay}
+          placement={placement}
+          {...restTooltipProps}
```
Comment (benjycui) [inline src/createSliderWithTooltip.jsx]: `{...restTooltipProps}` should be the first props.

Comment (zeakd) [inline src/createSliderWithTooltip.jsx]: is there any reason?

Comment (benjycui) [inline src/createSliderWithTooltip.jsx]: I afraid that user will override `visible`

Comment (benjycui) [inline src/createSliderWithTooltip.jsx]: This may cause some issues, if users are not familiar with implementation of rc-slider.

Comment (zeakd) [inline src/createSliderWithTooltip.jsx]: you are right. How about just destruct visible from tipProps? IMO ordering sensitive props makes it hard to maintain.

Comment (paranoidjk) [inline src/createSliderWithTooltip.jsx]: @zeakd actually, many of our code rely on props ordering to control the props override logic

Comment (zeakd) [inline src/createSliderWithTooltip.jsx]: I see. I updated it

File: src/createSliderWithTooltip.jsx [hunk 2]
```diff
@@ -26,13 +28,25 @@ export default function createSliderWithTooltip(Component) {
...
-      const { tipFormatter } = this.props;
+      const {
+        tipFormatter,
+        tipProps,
+      } = this.props;
+
+      const {
+        prefixCls = 'rc-slider-tooltip',
+        overlay = tipFormatter(value),
+        placement = 'top',
```
Comment (paranoidjk) [inline src/createSliderWithTooltip.jsx]: I think `placement` should be one of restTooltipProps, as for the defaultValue, just use:

```tsx
placement="top"
{...restTooltipProps}
overlay={tipFormatter(value)}	
prefixCls={prefixCls}
overlay={overlay}
```

Comment (zeakd) [inline src/createSliderWithTooltip.jsx]: Sorry, I have two question.
1. what is difference? `placement` is already have default value. is it convention?
2. how about `prefixCls`? I think `prefixCls` should have same form with `placement` in your suggestion. like 
```
placement='top'
prefixCls='rc-slider-tooltip'
{...restTooltipProps}
overlay={tipFormatter(value)}
overlay={overlay} 
visible={!disabled && (this.state.visibles[index] || dragging)}
key={index}
```

Comment (paranoidjk) [inline src/createSliderWithTooltip.jsx]: @zeak 

The difference is maintainability. But i think right now is ok. I will merge this and release a minor version. Thanks.

**Comment by paranoidjk** [comment]
@zeakd  can you rebase master and resolve the conflict?

**Comment by zeakd** [comment]
Oh there is new feature `handleStyle`. I'll do it

**Pull request merged**